tt_metal/impl/device/device_pool.cpp

```diff
@@ -371,13 +374,97 @@ bool DevicePool::close_device(chip_id_t device_id) {
     return pass;
 }
 
+void DevicePool::close_devices(std::vector<Device *> devices) {
+    // Sync and close all tunnels needed to support <devices>
+    bool is_galaxy = tt::Cluster::instance().is_galaxy_cluster();
+
+    // Ordered, because we need to shutdown tunnels from the farthest to the closest.
+    std::vector<chip_id_t> devices_to_close;
+
+    // Loop over all devices and add remote devices to devices_to_close
+    // For Galaxy if an mmio device's tunnels are being closed, close the mmio device as well
```

Comment:
I don't want this function to do anything too smart... I would want it to tell me that I can close these devices, or I can't -- not close any extra devices I don't explicitly want it to. If existing code relies on this behavior, can we add a flag to auto-close?